test-jest/cartesian/Scatter.spec.tsx

```diff
@@ -0,0 +1,108 @@
+
+describe('<Scatter />', () => {
+  const data = [
+    { cx: 10, cy: 50, size: 64, payload: { x: 12, y: 23, z: 78 } },
+    { cx: 50, cy: 50, size: 64, payload: { x: 12, y: 23, z: 78 } },
+    { cx: 90, cy: 50, size: 64, payload: { x: 12, y: 23, z: 78 } },
+    { cx: 130, cy: 50, size: 64, payload: { x: 12, y: 23, z: 78 } },
+    { cx: 170, cy: 50, size: 64, payload: { x: 12, y: 23, z: 78 } },
+  ];
+
+  test('Render symbols in a simple Scatter', () => {
+    const { container } = render(
+      <Surface width={500} height={500}>
+        <Scatter points={data as any} />
+      </Surface>,
+    );
+
+    expect(container.querySelectorAll('.recharts-scatter-symbol')).toHaveLength(data.length);
+  });
+
+  test("Don't render any symbols when data is empty", () => {
+    const { container } = render(
+      <Surface width={500} height={500}>
+        <Scatter points={[] as any} />
+      </Surface>,
+    );
+
+    expect(container.querySelectorAll('.recharts-scatter-symbol')).toHaveLength(0);
+  });
+
+  test('Render customized symbols when shape is set to be a ReactElement', () => {
+    const CustomizedShape = ({ cx, cy }: { cx: number; cy: number }) => (
+      <circle cx={cx} cy={cy} r={5} className="customized-shape" />
+    );
+    const { container } = render(
+      <Surface width={500} height={500}>
+        <Scatter isAnimationActive={false} shape={<CustomizedShape cx={0} cy={0} />} points={data as any} />
+      </Surface>,
+    );
+
+    expect(container.querySelectorAll('.customized-shape')).toHaveLength(data.length);
+  });
+
+  test('Render customized symbols when shape is set to be a function', () => {
+    const renderCustomizedShape = ({ cx, cy }: { cx: number; cy: number }) => (
+      <circle cx={cx} cy={cy} r={5} className="customized-shape" />
+    );
+    const { container } = render(
+      <Surface width={500} height={500}>
+        <Scatter isAnimationActive={false} shape={renderCustomizedShape} points={data as any} />
+      </Surface>,
+    );
+
+    expect(container.querySelectorAll('.customized-shape')).toHaveLength(data.length);
+  });
+
+  test('Render customized line when line is set to be a ReactElement', () => {
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars, react/prop-types
+    const CustomizedLine = ({ points }: { points: any }) => <path d="M0,0L200,200" className="customized-line" />;
+    const { container } = render(
+      <Surface width={500} height={500}>
+        <Scatter isAnimationActive={false} line={<CustomizedLine points={data as any} />} points={data as any} />
+      </Surface>,
+    );
+
+    expect(container.querySelectorAll('.customized-line')).toHaveLength(1);
+  });
+
+  test('Render customized line when line is set to be a function', () => {
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const renderCustomizedLine = ({ points }: { points: any }) => <path d="M0,0L200,200" className="customized-line" />;
```

Comment:
Please generally do not use any if possible. You can infer the type automatically.

Reply:
I think it's ok in test code.

Reply:
Even in tests, `any` has a chance to make things happen that we don't expect. Unless there is an explicit reason (testing if something breaks with wrong props passed, there is no type in the code itself) we should stick to as strict types as we can. Maybe we can say if we use `any` we have to comment above it as to why we did